docs/source/format/Intro.rst

```diff
@@ -0,0 +1,510 @@
+
+Sparse union
+^^^^^^^^^^^^
+
+A sparse union has the same structure as a dense union, with the omission of the offsets
+array. In this case, the child arrays are each equal in length to the length of the union.
+
+
+.. figure:: ./images/sparse-union-diagram.svg
+   :alt: Diagram is showing the difference between the sparse union data type
+         presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for sparse union data type.
+
+Dictionary Encoded Layout
+=========================
+
+Dictionary encoding can be effective when you have data with many repeated values.
+The values are represented by integers referencing a dictionary usually consisting of
+unique values.
+
+.. figure:: ./images/dictionary-diagram.svg
+   :alt: Diagram is showing the difference between the dictionary data type
+         presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for dictionary data type.
+
+Run-End Encoded Layout
+======================
+
+Run-end encoding is well-suited for representing data containing sequences of the
+same value. These sequences are called runs. Run-end encoded array has no buffers
+by itself, but has two child arrays:
+
+*  **Run ends array:** holds the index in the array where each run ends.
+*  **Values array:** the actual values without repetitions.
+
+.. figure:: ./images/ree-diagram.svg
+   :alt: Diagram is showing the difference between the run-end encoded data
+         type presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for run-end encoded data type.
+
+.. link to All types overview https://github.com/apache/arrow/issues/14752
+
+Extension Types
+===============
+
+In case the system or application needs to extend standard Arrow data types with
+custom semantics this is enabled by defining extension types or user-defined types.
+
+For example:
+
+* Universally unique identifiers (UUID) can be represented as a FixedSizeBinary type
+* Trading time can be represented as a Timestamp with metadata indicating the market
+  trading calendar
+
+Extension types can be defined by annotating any of the built-in Arrow data types
+(the “storage type”) with a custom type name and optional serialized representation
+(``'ARROW:extension:name'`` and ``'ARROW:extension:metadata'`` keys in the Field
+metadata structure).
+
+.. seealso::
+   The :ref:`format_metadata_extension_types` documentation.
+
+Canonical Extension Types
+-------------------------
+
+It is beneficial to share the definitions of well-known extension types so as to
+improve interoperability between different systems integrating Arrow columnar data.
+For this reason canonical extension types are defined in Arrow itself.
+
+Examples:
+
+* Fixed and variable shape tensor
+
+  - :ref:`fixed_shape_tensor_extension`
+  - :ref:`variable_shape_tensor_extension`
```

Comment:
This list can be updated now with json and uuid (although it maybe also raises the question if we just want to refer to other doc page, to avoid having to keep both in sync)